### PR TITLE
Feature/gridedit 611 snap polygon to land boundary

### PR DESCRIPTION
**Pull request by BillSenior**

### CODE REVIEW

File: libs/MeshKernel/include/MeshKernel/Polygons.hpp [hunk 1]
```diff
@@ -63,6 +67,12 @@ namespace meshkernel
         /// @return The new offset polygon
         [[nodiscard]] Polygons OffsetCopy(double distance, bool innerAndOuter) const;
 
+        /// @brief Snap the polygon to the land boundary
+        ///
+        /// The polygon points are snapped to the closest point on the land boundary.
+        /// @param[in] landBoundary The land boundary to which the polygon should be snapped.
+        void SnapToLandBoundary(const LandBoundary& landBoundary);
```
Comment (lucacarniato) [inline libs/MeshKernel/include/MeshKernel/Polygons.hpp:74]: I think we also need to select the polygon indices for selecting the part of the polygon that will be snapped

**Review by lucacarniato — CHANGES_REQUESTED** [review]

File: libs/MeshKernel/include/MeshKernel/Polygons.hpp (hunk 1 above)
Comment (BillSenior) [reply libs/MeshKernel/include/MeshKernel/Polygons.hpp]: Added start and end index

File: libs/MeshKernel/src/LandBoundary.cpp [hunk 2]
```diff
@@ -12,29 +12,44 @@ void meshkernel::LandBoundary::FindNearestPoint(const Point& samplePoint,
...
     nearestPoint = samplePoint;
-    segmentStartIndex = constants::missing::intValue;
+    segmentStartIndex = constants::missing::uintValue;
     scaledDistanceToStart = -1.0;
 
-    // TODO better value for initial value and move to constants namespace
     minimumDistance = 9.0e33;
 
-    for (UInt i = 0; i < m_nodes.size() - 1; ++i)
+    if (samplePoint.IsValid())
```
Comment (lucacarniato) [inline libs/MeshKernel/src/LandBoundary.cpp:20]: to avoid nesting you could return here?

**Review by lucacarniato — CHANGES_REQUESTED** [review]

File: libs/MeshKernel/src/LandBoundary.cpp (hunk 2 above)
Comment (BillSenior) [reply libs/MeshKernel/src/LandBoundary.cpp:20]: I think here the comparison tool in github has made a mistake

Comment (BillSenior) [reply libs/MeshKernel/src/LandBoundary.cpp:20]: If  you are comparing the code 

21 for (Uint i = 0 ...)

to

20 if(samplePoint.IsValid

Comment (BillSenior) [reply libs/MeshKernel/src/LandBoundary.cpp:20]: now I see what you mean

Comment (BillSenior) [reply libs/MeshKernel/src/LandBoundary.cpp]: changed

**Review by lucacarniato — APPROVED** [review]

**Pull request merged**